### PR TITLE
Composite storage optimization

### PR DESCRIPTION
**Pull request by sschuberth**

### CODE REVIEW

**Review by oheger-bosch — DISMISSED** [review]
Looks good to me.

**Comment by oheger-bosch** [comment]
Hm, there are of course tests for the storage configuration, which need to be adapted.

**Comment by sschuberth** [comment]
> Hm, there are of course tests for the storage configuration, which need to be adapted.

True, I just finished doing that 😀

File: scanner/src/test/kotlin/ScanResultsStorageTest.kt [hunk 1]
```diff
@@ -127,49 +116,44 @@ class ScanResultsStorageTest : WordSpec({
...
                 val config = ScannerConfiguration(
-                    storages = mapOf("reader" to ClearlyDefinedStorageConfiguration("http://url")),
+                    storages = mapOf(
+                        "cdProd" to ClearlyDefinedStorageConfiguration("http://prod.cd"),
+                        "cdDev" to ClearlyDefinedStorageConfiguration("http://dev.cd")
+                    ),
                     storageReaders = listOf("nonExistingReader")
                 )
-                configureStorage(config)
+                ScanResultsStorage.configure(config) should beInstanceOf<CompositeStorage>()
```
Comment (oheger-bosch) [inline scanner/src/test/kotlin/ScanResultsStorageTest.kt:138]: I find it a bit confusing that you added a check here. The configure() function is expected to throw an exception, so the instanceOf check will never be executed. And if it was executed and failed, the resulting failure message would be misleading.

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**